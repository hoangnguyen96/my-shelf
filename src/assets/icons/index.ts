export { HeartIconFull, HeartIconOutline } from "./HeartIcon";
export { LogoMain } from "./Logo";
export { CheckIcon } from "./CheckIcon";
export { HomeIcon } from "./HomeIcon";
export { SearchIcon } from "./SearchIcon";
export { BookshelfIcon } from "./BookshelfIcon";
export { GiftIcon } from "./GiftIcon";
export { StarOutlineIcon } from "./StarOutlineIcon";
export { StarFullIcon } from "./StarFullIcon";
export { UploadIcon } from "./Upload";
export { EditProfileIcon } from "./EditProfileIcon";
