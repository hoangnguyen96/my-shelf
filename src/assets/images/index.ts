export { default as avatar } from "./avatar.webp";
export { default as book1 } from "./book1.webp";
export { default as previewAuthor } from "./preview-author.webp";
