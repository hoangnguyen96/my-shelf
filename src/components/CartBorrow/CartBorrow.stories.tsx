import { Meta, StoryFn } from "@storybook/react";
import { Box, ChakraProvider } from "@chakra-ui/react";
import theme from "@app/themes";
import CartBorrow from ".";
import { RouterContext } from "next/dist/shared/lib/router-context.shared-runtime";
import { mockRouter, mockSession } from "@app/mocks/storybook";
import { SessionProvider } from "next-auth/react";

const meta: Meta<typeof CartBorrow> = {
  component: CartBorrow,
  decorators: [
    (Story: StoryFn) => (
      <ChakraProvider theme={theme}>
        <RouterContext.Provider value={mockRouter}>
          <SessionProvider session={mockSession}>
            <Story />
          </SessionProvider>
        </RouterContext.Provider>
      </ChakraProvider>
    ),
  ],
};

export default meta;

const Template: StoryFn<typeof CartBorrow> = () => (
  <CartBorrow
    title={"Donâ€™t Make Me think"}
    author={"Steve Krug, "}
    publicationYear={1900}
    rating={4.5}
    imgUrl="https://i.ibb.co/2Kn2kW2/book1.webp"
    createDate={"2024-10-15T08:32:57.347Z"}
    id={""}
    onReturnBook={() => {}}
  />
);

export const Default = Template.bind({});
Default.args = {};
