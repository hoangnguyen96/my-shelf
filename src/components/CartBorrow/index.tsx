"use client";

import { memo } from "react";
import { Box, Flex, Text } from "@chakra-ui/react";
import Image from "next/image";
import { formatDate } from "@app/utils";
import { Button } from "../common";

interface CartBorrowProps {
  id: string;
  title: string;
  author: string;
  publicationYear: number;
  rating: number;
  imgUrl: string;
  createDate: string;
  onReturnBook: (id: string) => void;
}

const CartBorrow = ({
  id,
  title,
  author,
  publicationYear,
  rating,
  imgUrl,
  createDate,
  onReturnBook,
}: CartBorrowProps) => {
  const handleReturnBook = () => {
    onReturnBook(id);
  };

  return (
    <Flex
      gap="25px"
      w={308}
      maxH={290}
      py="15px"
      pl="15px"
      pr="20px"
      bgColor="white"
      borderRadius="10px"
    >
      <Flex flexDirection="column" gap="5px" w={123}>
        <Image
          src={imgUrl}
          alt={title || "Image Cart borrowed!"}
          width={123}
          height={170}
          priority={true}
          fetchPriority="high"
        />
        <Text size="sm">{title}</Text>
        <Flex>
          <Text size="xs">{author},</Text>
          <Text size="xs">{publicationYear}</Text>
        </Flex>
        <Flex>
          <Text size="xs">{rating}/</Text>
          <Text size="xs" color="dark.70">
            5
          </Text>
        </Flex>
      </Flex>
      <Flex flexDirection="column" justifyContent="space-between">
        <Box>
          <Text size="md" mb="10px">
            Borrowed on
          </Text>
          <Text size="xs">{formatDate(new Date(createDate))}</Text>
        </Box>
        <Button
          data-testid="return-book"
          variant="outline"
          size="md"
          w={125}
          boxShadow="none"
          text="Return"
          onClick={handleReturnBook}
        />
      </Flex>
    </Flex>
  );
};

export default memo(CartBorrow);
