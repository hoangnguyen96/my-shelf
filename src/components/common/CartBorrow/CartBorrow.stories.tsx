import { Meta, StoryFn } from "@storybook/react";
import { ChakraProvider } from "@chakra-ui/react";
import theme from "@app/themes";
import CartBorrow from ".";
import { book1 } from "@app/assets/images";

const meta: Meta<typeof CartBorrow> = {
  component: CartBorrow,
  decorators: [
    (Story: StoryFn) => (
      <ChakraProvider theme={theme}>
        <Story />
      </ChakraProvider>
    ),
  ],
};

export default meta;

const Template: StoryFn<typeof CartBorrow> = () => (
  <CartBorrow
    title={"Donâ€™t Make Me think"}
    author={"Steve Krug, "}
    publicationYear={1900}
    rating={4.5}
    imgUrl={`${book1}`}
    createDate={"11 Mar 2023 09:00 AM"}
    id={""}
    onReturnBook={() => {}}
  />
);

export const Default = Template.bind({});
Default.args = {};
