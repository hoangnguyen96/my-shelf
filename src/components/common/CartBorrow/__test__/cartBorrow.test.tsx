import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import CartBorrow from "..";
import { book1 } from "@app/assets/images";

describe("CartBorrow", () => {
  it("Should render correctly snapshot", () => {
    expect(
      render(
        <CartBorrow
          title={"Donâ€™t Make Me think"}
          author={"Steve Krug, "}
          publicationYear={1900}
          rating={4.5}
          imgUrl={`${book1}`}
          createDate={"11 Mar 2023 09:00 AM"}
          id={""}
          onReturnBook={() => {}}
        />
      )
    ).toMatchSnapshot();
  });
});
