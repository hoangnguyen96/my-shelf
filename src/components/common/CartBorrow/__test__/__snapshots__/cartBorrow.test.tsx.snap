// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartBorrow Should render correctly snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-415p7i"
      >
        <div
          class="css-n736ek"
        >
          <img
            alt="Don't make me think"
            data-nimg="1"
            decoding="async"
            height="40"
            loading="lazy"
            src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
            srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
            style="color: transparent;"
            width="40"
          />
          <p
            class="chakra-text css-0"
          >
            Don’t Make Me think
          </p>
          <div
            class="css-k008qs"
          >
            <p
              class="chakra-text css-0"
            >
              Steve Krug,
            </p>
            <p
              class="chakra-text css-0"
            >
              2000
            </p>
          </div>
          <div
            class="css-k008qs"
          >
            <p
              class="chakra-text css-0"
            >
              4.5/
            </p>
            <p
              class="chakra-text css-c5z1pm"
            >
              5
            </p>
          </div>
        </div>
        <div
          class="css-1b6n4o1"
        >
          <div
            class="css-0"
          >
            <p
              class="chakra-text css-1r2f04i"
            >
              Borrowed on
            </p>
            <p
              class="chakra-text css-0"
            >
              11 Mar 2023 09:00 AM
            </p>
          </div>
          <button
            class="chakra-button css-16ltlly"
            type="button"
          >
            Return
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-415p7i"
    >
      <div
        class="css-n736ek"
      >
        <img
          alt="Don't make me think"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <p
          class="chakra-text css-0"
        >
          Don’t Make Me think
        </p>
        <div
          class="css-k008qs"
        >
          <p
            class="chakra-text css-0"
          >
            Steve Krug,
          </p>
          <p
            class="chakra-text css-0"
          >
            2000
          </p>
        </div>
        <div
          class="css-k008qs"
        >
          <p
            class="chakra-text css-0"
          >
            4.5/
          </p>
          <p
            class="chakra-text css-c5z1pm"
          >
            5
          </p>
        </div>
      </div>
      <div
        class="css-1b6n4o1"
      >
        <div
          class="css-0"
        >
          <p
            class="chakra-text css-1r2f04i"
          >
            Borrowed on
          </p>
          <p
            class="chakra-text css-0"
          >
            11 Mar 2023 09:00 AM
          </p>
        </div>
        <button
          class="chakra-button css-16ltlly"
          type="button"
        >
          Return
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
